Phonebook.Backend/Phonebook.Backend.PictureService/Phonebook.Backend.PictureService/Controllers/EmployeePicturesController.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Cors;
+using Phonebook.Backend.PictureService.Helpers;
+
+namespace Phonebook.Backend.PictureService.Controllers
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    [Authorize]
+    [EnableCors("AllowDomainList")]
+    [Route("/")]
+    public class EmployeePictureController : Controller
+    {
+        /// <summary>
+        /// Upload an employee picture.
+        /// </summary>
+        /// <remarks>Content has to be formatted as "image/jpeg"</remarks>
+        /// See <see cref="UserController.WhoAmI()"/> to get the currently logged in user.
+        /// <param name="id">The id of the user.</param>
+        /// <param name="file">A "multipart"</param>
+        /// <returns>
+        /// A status code indicating the success of the action.
+        /// </returns>
+        /// <permission cref="System.Security.PermissionSet">Everyone can access this method.</permission>
+        /// <response code="200">Ok</response>
+        /// <response code="400">No Payload</response>
+        /// <response code="400">Content malformated</response>
+        /// <response code="401">User not authorized</response>
+        /// <response code="500">Internal Server Error</response>
+        [HttpPost("{id}")]
+        [RequestSizeLimit(2147483648)]
+        public async Task<IActionResult> UploadPicture(string id, IFormFile file)
+        {
+            if (file == null || file.Length == 0)
+            {
+                return BadRequest("No Payload.");
+            }
+            var user = HttpContext.User;
+            if (user.Identity.Name.Split("\\")[1] != id)
+            {
+                return StatusCode(401, "User not authorized.");
+            }
+            if (

```

Comment:
- please make this If more central. 
- put common formats before not known formats. This will burst our performance.

Reply:
is there something like Map in C#? We could check for each occurrence in the Map.

Reply:
I think to sort the ContentTypes by the probability of their occurrence. this maybe can improve the performance a little bit.
This would make sense because of the if will not try any other one when 1 expression matches.

Reply:
Is there a way to do this at runtime level? Like load a list of supported types and then try them according to their position. If one matches successfully increase their rank in the list.

Reply:
maybe this is a feature for the images lib

Reply:
Could be yeah, because it is providing the conversion, this code does not know which types are supported.

Reply:
We could use a switch statement as well as it provides log(n) complexity

Reply:
The lib should provide a "public bool IsFormatSuported("formatstring")" or something similar.
Can you please file an issue there for that?

Reply:
Mhm, no. This pretty use-case-specific. 
The Method would only be useful for internet-based applications. Applications operating on command line level would need a function for file-type-extensions. 
Which would add a lot of complexity, aka. would add to the runtime complexity.

Reply:
Changed the order and of the allowed Content types. More famous Content types first.